zzip/__string.h

```diff
@@ -1,6 +1,8 @@
 #ifndef __ZZIP_INTERNAL_STRING_H
 #define __ZZIP_INTERNAL_STRING_H
 
+#include <stdlib.h>
```

Comment:
I was not sure if this header can be added unconditionally or not. Without it the build would fail on Solaris with
```
../../zziplib/zzip/__string.h:44:19: error: implicit declaration of function 'malloc' [-Werror=implicit-function-declaration]
         char* r = malloc(len + 1);
                   ^
```